src/qinfer/utils.py

```diff
@@ -454,6 +454,48 @@ def pretty_time(secs, force_h=False, force_m=False):
 def safe_shape(arr, idx=0, default=1):
     shape = np.shape(arr)
     return shape[idx] if idx < len(shape) else default
+    
+def join_struct_arrays(arrays):
+    """
+    Takes a list of possibly structured arrays, concatenates their
+    dtypes, and returns one big array with that dtype. Does the 
+    inverse of ``separate_struct_array``.
+    
+    :param list arrays: List of ``np.ndarray``s
+    """
+    # taken from http://stackoverflow.com/questions/5355744/numpy-joining-structured-arrays
+    sizes = np.array([a.itemsize for a in arrays])
+    offsets = np.r_[0, sizes.cumsum()]
+    shape = arrays[0].shape
```

Comment:
``arrays`` is being called with the result of ``map`` in line 238 of ``domains.py``, but ``map`` is not indexible on Python 3. I'd suggest calling ``list(map(...))`` above since that also clears the Py3 warning that's blocking the Travis build.